blivet/osinstall.py

```diff
@@ -1194,6 +1195,139 @@ def write(self):
+
+    @property
+    def boot_device(self):
+        dev = None
+        root_device = self.mountpoints.get("/")
+
+        dev = self.mountpoints.get("/boot", root_device)
+        return dev
+
+    @property
+    def default_boot_fstype(self):
+        """The default filesystem type for the boot partition."""
+        if self._default_boot_fstype:
+            return self._default_boot_fstype
+
+        fstype = None
+        if self.bootloader:
+            fstype = self.boot_fstypes[0]
+        return fstype
+
+    def set_default_boot_fstype(self, newtype):
+        """ Set the default /boot fstype for this instance.
+
+            Raise ValueError on invalid input.
+        """
+        log.debug("trying to set new default /boot fstype to '%s'", newtype)
+        # This will raise ValueError if it isn't valid
+        self._check_valid_fstype(newtype)
+        self._default_boot_fstype = newtype
+
+    def set_up_bootloader(self, early=False):
+        """ Propagate ksdata into BootLoader.
+
+            :keyword bool early: Set to True to skip stage1_device setup
+
+            :raises BootloaderError: if stage1 setup fails
+
+            If this needs to be run early, eg. to setup stage1_disk but
+            not stage1_device 'early' should be set True to prevent
+            it from raising BootloaderError
+        """
+        if not self.bootloader or not self.ksdata:
+            log.warning("either ksdata or bootloader data missing")
+            return
+
+        if self.bootloader.skip_bootloader:
+            log.info("user specified that bootloader install be skipped")
+            return
+
+        # Need to make sure bootDrive has been setup from the latest information
+        self.ksdata.bootloader.execute(self, self.ksdata, None)
+        self.bootloader.stage1_disk = self.devicetree.resolve_device(self.ksdata.bootloader.bootDrive)
+        self.bootloader.stage2_device = self.boot_device
+        if not early:
+            self.bootloader.set_stage1_device(self.devices)
+
+    @property
+    def boot_disk(self):
+        disk = None
+        if self.ksdata:
+            spec = self.ksdata.bootloader.bootDrive
+            disk = self.devicetree.resolve_device(spec)
+        return disk
+
+    @property
+    def bootloader_device(self):
+        return getattr(self.bootloader, "stage1_device", None)
+
+    @property
+    def boot_fstypes(self):
+        """A list of all valid filesystem types for the boot partition."""
+        fstypes = []
+        if self.bootloader:
+            fstypes = self.bootloader.stage2_format_types
+        return fstypes
+
+    def get_fstype(self, mountpoint=None):
+        """ Return the default filesystem type based on mountpoint. """
+        fstype = super().get_fstype(mountpoint=mountpoint)
+
+        if not mountpoint:
+            # just return the default
+            pass
+        elif mountpoint == "/boot":
+            fstype = self.default_boot_fstype
+
```

Comment:
I wasn't actually sure if that check was over-kill or necessary, so I'll get rid of it.